src/platform_impl/linux/mod.rs

```diff
@@ -754,65 +738,26 @@ impl<T: 'static> EventLoop<T> {
             );
         }
 
-        #[cfg(x11_platform)]
-        if attributes.forced_backend == Some(Backend::X) {
-            // TODO: Propagate
-            return EventLoop::new_x11_any_thread().unwrap();
-        }
-
+        // NOTE: Wayland first because of X11 could be present under wayland as well.
         #[cfg(wayland_platform)]
-        if attributes.forced_backend == Some(Backend::Wayland) {
-            // TODO: Propagate
-            return EventLoop::new_wayland_any_thread().expect("failed to open Wayland connection");
-        }
-
-        if let Ok(env_var) = env::var(BACKEND_PREFERENCE_ENV_VAR) {
-            match env_var.as_str() {
-                "x11" => {
-                    // TODO: propagate
-                    #[cfg(x11_platform)]
-                    return EventLoop::new_x11_any_thread()
-                        .expect("Failed to initialize X11 backend");
-                    #[cfg(not(x11_platform))]
-                    panic!("x11 feature is not enabled")
-                }
-                "wayland" => {
-                    #[cfg(wayland_platform)]
-                    return EventLoop::new_wayland_any_thread()
-                        .expect("Failed to initialize Wayland backend");
-                    #[cfg(not(wayland_platform))]
-                    panic!("wayland feature is not enabled");
-                }
-                _ => panic!(
-                    "Unknown environment variable value for {BACKEND_PREFERENCE_ENV_VAR}, try one of `x11`,`wayland`",
-                ),
-            }
```

Comment:
Is there any reason why you're getting rid of this mechanism for selecting the Linux backend? According to the users that I've spoken with it's a useful tool.

Reply:
You have a standard mechanism of unsetting `WAYLAND_DISPLAY`, the mechanism is error prone, because if you use other crates, it obviously won't work for them.

The real way to do is unset `WAYLAND_DISPLAY` or `DISPLAY`.

Reply:
I suppose so, but as this is a user-facing change it should be explicitly mentioned in the `CHANGELOG.md`, alongside the alternative you gave.